### PR TITLE
Update rust edition

### PR DESCRIPTION
**Pull request by NikVolf**

### CODE REVIEW

File: trie-db/src/node_codec.rs [hunk 1]
```diff
@@ -16,8 +16,8 @@
 //! to parametrize the hashes used in the codec.
 
 use hash_db::MaybeDebug;
-use node::{Node, NodePlan};
-use ChildReference;
+use crate::node::{Node, NodePlan};
+use crate::ChildReference;
 #[cfg(feature = "std")]
 use std::borrow::Borrow;
```
Comment (ordian) [inline trie-db/src/node_codec.rs]: can we `use crate::core_::borrow::Borrow;` here and below?

File: hash256-std-hasher/src/lib.rs [hunk 2]
```diff
@@ -14,15 +14,14 @@
...
 #![cfg_attr(not(feature = "std"), no_std)]
 
-#[macro_use]
-extern crate crunchy;
-
 #[cfg(feature = "std")]
 use std::hash;
 
 #[cfg(not(feature = "std"))]
 use core::hash;
```
Comment (ordian) [inline hash256-std-hasher/src/lib.rs]: can we `use crate::core_::hash;` here and above?

File: trie-db/src/trie_codec.rs [hunk 3]
```diff
@@ -32,7 +32,7 @@ use crate::{
 	nibble_ops::NIBBLE_LENGTH, node::{Node, NodeHandle, NodeHandlePlan, NodePlan, OwnedNode},
 };
 
-use	::core_::{
+use	crate::core_::{
```
Comment (ordian) [inline trie-db/src/trie_codec.rs]:
```suggestion
use crate::core_::{
```

**Comment by NikVolf** [comment]
hey @ordian I think those changes out of scope, I just migrate to 2018 edition here

So we can add another issue to sort proper namespace management - #54

**Pull request merged**